onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selector_action_transformer.cc

```diff
@@ -20,7 +20,7 @@ void SplitQDQRules(SelectorActionRegistry& qdq_selector_action_registry) {
   const std::string action_name{"dropSplitQDQ"};
   std::unique_ptr<Action> action = std::make_unique<QDQ::SplitReplaceWithQuant>();
 #if !defined(ORT_MINIMAL_BUILD)
-  std::unique_ptr<NodeSelector> selector = std::make_unique<QDQ::OutputVariadicSelector>();
```

Comment:
Note: The OutputVariadicSelector did not check that all quantization params matched. Additionally, the OutputVariadicSelector did not consider a Split operator with an optional split input an actual QDQ node group (missed optimization opportunity for CPU EP).

Reply:
Should the OutputVariadicSelector do that check? Maybe that's where the check on everything matching would need to be configurable.

Reply:
Both the `OutputVariadicSelector` and `InputVariadicSelector` classes use the same `VariadicNodeGroupSelector` class underneath the hood, which assumes the inputs and outputs are all variadic. 

I opted to give Split its own selector class because making the `VariadicNodeGroupSelector` configurable seemed to add unnecessary complexity that is not needed for users of `InputVariadicSelector`.